.github/workflows/installer.yml

```diff
@@ -24,7 +24,6 @@ jobs:
       matrix:  # just a vector in this case
         make_env: [
           IPC_VERSION=,  # default version
-          IPC_VERSION=IPC4 UNSIGNED_list= SIGNED_list='tgl tgl-h cnl icl jsl',
```

Comment:
Need to leave this in atm as CI and most developers only have CAVS platforms for test and dev work on IPC4.

Reply:
This removes IPC4 compilation for all platforms without replacing them with any IPC4 platform. So this is effectively turning off IPC4 compilation entirely which makes no sense and makes even less sense in an IPC4 pull request. This looks like trying to hide compilation issues.

As Liam mentioned, IPC4 testing is currently performed on IPC3 platforms. Even if it were not, this type of "poor man's randconfig" is useful because it can find logical Kconfig bugs. You can never have too much randconfig.

Reply:
> This removes IPC4 compilation for all platforms without replacing them with any IPC4 platform. So this is effectively turning off IPC4 compilation entirely which makes no sense and makes even less sense in an IPC4 pull request. This looks like trying to hide compilation issues.

@marc-hb I think that you are missing the point of this PR. The current IPC4 implementation has major design flaws that require code refactor. It is required to move forward and prepare us for Intel MTL platform support that officially will be released with IPC4 protocol. None of the cAVS targets has planned date to switch to support IPC4 protocol and all the cAVS releases are currently going with IPC3 - this is not a priority for now and cannot be blocking for MTL release.

We can check if there are some quick w/a that can be applied to sustain cAVS compilation with IPC4 for now, but it will be replaced with MTL IPC4 CI as soon as MTL support will be upstream. The cAVS IPC4 configuration support will require adaptation to incoming changes and depending on available bandwidth might require to be temporary turned off. 

The bottom line is that we cannot put MTL SOF + Zephyr IPC4 release at risk in favor of maintaining cAVS IPC4 support. If you have any problem with that then please contact me directly.

Reply:
@marc-hb I indeed introduced some errors in Friday's PR cleanup attempt and now fixed them.
I assure you I'm not attempting to hide compilation issues, but to cleanup ipc4 usage.
It's a bit disappointing to me that you read it this way, hopefully it won't discourage others from improving this FW.

From information I have cavs + ipc4 is not a valid configuration.
MTL + IPC4 is a valid one but I accept Liam's comment about MTL not being available to developers so in latest version I'm providing changes which should satisfy IPC4 + cavs config for the time being.

Please let me know if you have further comments.

Reply:
> The current IPC4 implementation has major design flaws that require code refactor. [...]
> The bottom line is that we cannot put MTL SOF + Zephyr IPC4 release at risk in favor of maintaining cAVS IPC4 support.

I think any refactor that breaks cAVS IPC4 is fine as long as it compiles IPC4 code in CI "somewhere" else instead[*]. Any configuration can do. Merging code that is never compiled in CI does not move anything forward, it regresses and slows down everyone, breaks git bisect etc. If a huge jump is required then it can be submitted as a single giant commit.

Maybe I'm missing something here but... are we really discussing whether every git commit should compile?


[*] Other people actively developing IPC4 on IPC3 platforms may disagree.